File: Csharp/VueClient/Models/Menu.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VueClient.Models;

[Table("menu")]
public class Menu
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [StringLength(100)]
    [Column("nom")]
    public string Nom { get; set; } = string.Empty;

    [Required]
    [Column("prix", TypeName = "decimal(10,2)")]
    public decimal Prix { get; set; }

    [Column("description", TypeName = "text")]
    public string? Description { get; set; }

    [StringLength(255)]
    [Column("image")]
    public string? Image { get; set; }

    [Required]
    [Column("id_burger")]
    [ForeignKey("Burger")]
    public int IdBurger { get; set; }

    [Required]
    [Column("id_boisson")]
    [ForeignKey("ComplementBoisson")]
    public int IdBoisson { get; set; }

    [Required]
    [Column("id_frite")]
    [ForeignKey("ComplementFrite")]
    public int IdFrite { get; set; }

    [Column("is_archive")]
    public bool IsArchive { get; set; } = false;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public virtual Burger Burger { get; set; } = null!;
    public virtual Complement ComplementBoisson { get; set; } = null!;
    public virtual Complement ComplementFrite { get; set; } = null!;
}

File: symfony/src/Entity/Menu.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'menu')]
class Menu
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: 'IDENTITY')]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(type: 'string', length: 100, nullable: false)]
    private string $nom;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2, nullable: false)]
    private string $prix;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\ManyToOne(targetEntity: Burger::class)]
    #[ORM\JoinColumn(name: 'id_burger', referencedColumnName: 'id', nullable: false, onDelete: 'RESTRICT')]
    private Burger $burger;

    #[ORM\ManyToOne(targetEntity: Complement::class)]
    #[ORM\JoinColumn(name: 'id_boisson', referencedColumnName: 'id', nullable: false, onDelete: 'RESTRICT')]
    private Complement $boisson;

    #[ORM\ManyToOne(targetEntity: Complement::class)]
    #[ORM\JoinColumn(name: 'id_frite', referencedColumnName: 'id', nullable: false, onDelete: 'RESTRICT')]
    private Complement $frite;

    #[ORM\Column(name: 'is_archive', type: 'boolean', nullable: false)]
    private bool $isArchive = false;

    #[ORM\Column(name: 'created_at', type: 'datetime', nullable: false)]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(name: 'updated_at', type: 'datetime', nullable: false)]
    private \DateTimeInterface $updatedAt;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->updatedAt = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getPrix(): string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): self
    {
        $this->prix = $prix;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getBurger(): Burger
    {
        return $this->burger;
    }

    public function setBurger(Burger $burger): self
    {
        $this->burger = $burger;
        return $this;
    }

    public function getBoisson(): Complement
    {
        return $this->boisson;
    }

    public function setBoisson(Complement $boisson): self
    {
        $this->boisson = $boisson;
        return $this;
    }

    public function getFrite(): Complement
    {
        return $this->frite;
    }

    public function setFrite(Complement $frite): self
    {
        $this->frite = $frite;
        return $this;
    }

    public function isArchive(): bool
    {
        return $this->isArchive;
    }

    public function setIsArchive(bool $isArchive): self
    {
        $this->isArchive = $isArchive;
        return $this;
    }

    public function getCreatedAt(): \DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): self
    {
        $this->createdAt = $createdAt;
        return $this;
    }

    public function getUpdatedAt(): \DateTimeInterface
    {
        return $this->updatedAt;
    }

    public function setUpdatedAt(\DateTimeInterface $updatedAt): self
    {
        $this->updatedAt = $updatedAt;
        return $this;
    }
}
